### DISCUSSION TITLE
How to return to 'main menu' after multi-level SelectionPrompt

### DISCUSSION THREAD

**Discussion by alexjamesbrown** [spectreconsole/spectre.console#1007] [Q&A] [answered]
I'm trying to create a multi-level menu with Spectre.Console that returns to the 'main menu' after completion

    Menu 1
    
    > Option 1
      Option 2

When I go into Option 1, I get the 'sub menu'

    Sub menu for Option 1
    
    > A
      B
      < Back

I'm trying to get it to go back to the main menu if I select `< Back` - but also after the work in `A` is complete.

My code so far (feels quite clumsy with these nested switch statements)

    var mainMenuResult = AnsiConsole.Prompt
    (
        new SelectionPrompt<string>()
            .Title("Menu 1")
            .AddChoices
            (
                "Option 1",
                "Option 2"
            )
    );
    
    switch (mainMenuResult)
    {
        case "Option 1":
            var option1MenuResult = AnsiConsole.Prompt
            (
                new SelectionPrompt<string>()
                    .Title("Sub menu for Option 1")
                    .AddChoices
                    (
                        "A",
                        "B",
                        "< Back"
                    )
            );
    
            switch (option1MenuResult)
            {
                case "A":
                    Console.WriteLine("You selected A");
                    // doing some work
                    // + go back to the main menu when complete
                    break;
    
                case "B":
                    Console.WriteLine("You selected B");
                    // doing some work
                    // + go back to the main menu when complete
                    break;
    
                case "< Back":
                    break;
            }
    
            break;
    
        case "Option 2":
        case "Option 3":
            // omitted for brevity
            break;
    }

**Comment by phil-scott-78** [comment, marked as answer]
This has come up before in another issue. There isn't any built in support for this, and I think people have been looking for work arounds to get the selection prompt to kinda work.

I think if I had the need I'd look at the source of the selection prompt as the basis of a new control designed specifically to do this rather than trying to get the selection prompt to do something it wasn't designed for.

Lots more features could be baked in for a consistent API to be reused too

**Reply by MrHinsh** [reply]
Any updates on a "menu" prompt?

**Reply by patriksvensson** [reply]
No updates

**Comment by MelGrubb** [comment]
Resurrecting this thread since this is where my search brought me. I created multiple different "Menu" classes with .Show methods on them. Each one looks basically like this.

```csharp
internal class MainMenu
{
    public static int? Show()
    {
        int? result = null;

        while (result == null)
        {
            AnsiConsole.Clear();
            AnsiConsole.Write(new FigletText("Main Menu").Centered());
            AnsiConsole.Write(new Rule());

            var prompt = AnsiConsole.Prompt(
                new SelectionPrompt<string>()
                    .Title("Choose a function")
                    .AddChoices("Foo", "Bar", "Exit"));

            switch (prompt)
            {
                case "Foo":
                    result = FooMenu.Show();
                    break;
                case "Bar":
                    result = BarMenu.Show();
                    break;
                default:
                    result = 0;
                    break;
            }
        }

        return result;
    }
}
```

Each sub-menu has a "Return" option, which returns null from the method, and an "Exit" option, which returns 0. So, if I go to the "Bar" menu, and then select "Return", `Bar.Show` returns null, and the loop in `MainMenu` simply displays again. If some option I selected from `BarMenu` had returned a non-null value, then the main menu loop would exit. This is just a basic proof of concept at this point, but maybe it will help someone else.

tl;dr - Put your menu in a loop and call some separate sub-menu method in the same class or some other. Look for meaningful return values and loop the menu until you see them.
